examples/llm_plus_gnn/glem.py

```diff
@@ -0,0 +1,372 @@
+    dataset = PygNodePropPredDataset(f'ogbn-{dataset_name}', root=root)
+    split_idx = dataset.get_idx_split()
+    data = dataset.data
+
+    tag_dataset = TAGDataset(root, dataset, hf_model,
+                             token_on_disk=token_on_disk)
+    text_dataset = tag_dataset.to_text_dataset()
+    print(tag_dataset.num_classes, tag_dataset.raw_file_names)
+
+    num_classes = tag_dataset.num_classes
+    num_features = data.num_features
+    # =========================== LM Data split ===============================
+    split_idx = tag_dataset.get_idx_split()
+
+    # GLEM train with augmented data, mark original train data as gold data,
+    gold_idx = split_idx['train']
+    split_idx['valid']
+    test_idx = split_idx['test']
+
+    # randome sample pseudo labels nodes, generate their index
+    num_pseudo_labels = int(gold_idx.numel() * pl_ratio)
+    idx_to_select = torch.randperm(test_idx.numel())[:num_pseudo_labels]
+    pseudo_labels_idx = test_idx[idx_to_select]
+    train_idx = torch.cat(
+        (gold_idx, pseudo_labels_idx))  # augmented train_indx
+
+    print(f'train_idx: {train_idx.size(0)}, '
+          f'gold_idx: {gold_idx.size(0)}, '
+          f'pseudo labels ratio: {pl_ratio}, '
+          f'{train_idx.size(0)/gold_idx.size(0) - 1.0}')
+    gold_dataset = torch.utils.data.Subset(dataset=text_dataset,
+                                           indices=gold_idx)
+    train_dataset = torch.utils.data.Subset(dataset=text_dataset,
+                                            indices=train_idx)
+    # valid_dataset = torch.utils.data.Subset(dataset=text_dataset,
+    #                                         indices=valid_idx)
+
+    # ========================== LM Data Loader ===============================
+
+    print('Building language model dataloader...', end='-->')
+    from torch_geometric.loader import DataLoader
+    text_pretrain_loader = DataLoader(gold_dataset, batch_size=lm_batch_size,
+                                      drop_last=False, pin_memory=False,
+                                      shuffle=True)
+    text_train_loader = DataLoader(train_dataset, batch_size=lm_batch_size,
+                                   drop_last=False, pin_memory=False,
+                                   shuffle=True)
+    # text_val_loader = DataLoader(valid_dataset, batch_size=lm_batch_size * 4,
+    #                              drop_last=False, pin_memory=True,
+    #                              shuffle=False)
+    text_data_loader = DataLoader(text_dataset, batch_size=lm_batch_size * 4,
+                                  drop_last=False, pin_memory=False,
+                                  shuffle=False)
+    print('done')
+
+    # =========================== GNN Data Loader =============================
+    initial_memory = torch.cuda.memory_allocated()
+    data = data.to(device)
+
+    current_memory_1 = torch.cuda.max_memory_allocated()
+    gpu_usage = float(current_memory_1 - initial_memory) / 1073741824
```

Comment:
put comment explaining what the hardcoded value means

Reply:
1 GB = 1073741824 Bytes

Reply:
yes a comment like "# GPU usage, in GB" would be good